OpenEdXMobile/src/main/java/org/edx/mobile/util/ResourceUtil.java

```diff
@@ -35,12 +35,12 @@ public static CharSequence getFormattedString(@NonNull String pattern, @NonNull
         Phrase resourceString = Phrase.from(pattern);
         Set<String> keys = keyValMap.keySet();
         for (String key : keys) {
-            CharSequence val = keyValMap.get(key);
+            final CharSequence val = keyValMap.get(key);
             if (val == null) {
                 logger.warn(String.format("Value for key %s is null", key));
-                resourceString.put(key, "");
+                resourceString.putOptional(key, "");
```

Comment:
This shouldn't be made optional, and if there are usages that are implemented wrongly (that come under this PR) they should be fixed as well.